src/runtime/GlobalObjectBuiltinPromise.cpp

```diff
@@ -357,6 +357,42 @@ static Value builtinPromiseCatch(ExecutionState& state, Value thisValue, size_t
 }
 
 
+static Value builtinPromiseFinally(ExecutionState& state, Value thisValue, size_t argc, Value* argv, bool isNewExpression)
+{
+    auto strings = &state.context()->staticStrings();
+    if (!thisValue.isObject())
+        ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, strings->Promise.string(), false, strings->then.string(), "%s: not a Promise object");
+    Object* thisObject = thisValue.toObject(state);
```

Comment:
I'd rather use `asObject`.

Reply:
This request is still valid.

Reply:
If you use asObject(), then the following tests will fail with an inner assertion, as:
`escargot: src/runtime/ValueInlines.h:528: Escargot::PointerValue* Escargot::Value::asPointerValue() const: Assertion isPointerValue() failed.`

**tests:**
```
assert.throws(TypeError, function() {
  Promise.prototype.finally.call(undefined);
}, 'undefined');

assert.throws(TypeError, function() {
  Promise.prototype.finally.call(null);
}, 'null');

```
and those are expecting the TypeError in that case, which case the isObject() call allows.

Reply:
I've checked out the PR and these are the result. First of all, let's simplify the test.

```js
Promise.prototype.finally.call(null)
```

Output:
```
Uncaught TypeError: Promise.then: not a Promise object:
function finally() { [native function] }  (-1:-1)
function call() { [native function] }  (-1:-1)
issue.js (2:1)
[Inferior 1 (process 27281) exited with code 03]
```

Whenever `thisValue` passes the `isObject()`condition, it can be safely converted to object with `asObject()`. I see no issue there.

Reply:
I'm gonna be honest, i cannot follow you right now. The PR uses `onFinally.isObject()`, which should and is throwing the TypeError, such as the test wants, and such as v8 handles it.
If i change into `asObject()`, as you suggest, the tests will inner assert fail on the `onFinally.asObject()` line with error message written above in my comment. That fail will be failing the whole testsuite, and is not corresponding to other similar engines in behaviour.

Reply:
My request was about to change the `toObject` into `asObject`, not the `isObject` into `asObject`.

Reply:
Then i know why i could keep up with. Gonna change it soon.